src/index.js

```diff
@@ -25,12 +25,15 @@ module.exports = function parseXml(xml, options = emptyObject) {
     Syntax = require('./lib/syntax');
   }
 
+  let pos = 0;
   if (xml[0] === '\uFEFF') {
     // Strip byte order mark.
-    xml = xml.slice(1);
+    pos = 1;
   }
 
-  xml = xml.replace(/\r\n/g, '\n'); // Normalize CRLF to LF.
+  if (!options.addOffsets) {
+    xml = xml.replace(/\r\n/g, '\n'); // Normalize CRLF to LF.
```

Comment:
The XML standard [requires that parsers behave as if `\r\n` is normalized to `\n`](https://www.w3.org/TR/2008/REC-xml-20081126/#sec-line-ends):

> XML parsed entities are often stored in computer files which, for editing convenience, are organized into lines. These lines are typically separated by some combination of the characters CARRIAGE RETURN (#xD) and LINE FEED (#xA).
>
> To simplify the tasks of applications, the XML processor must behave as if it normalized all line breaks in external parsed entities (including the document entity) on input, before parsing, by translating both the two-character sequence #xD #xA and any #xD that is not followed by #xA to a single #xA character.

If we skip this normalization when `options.addOffsets` is true, then the parsed document tree may contain carriage return characters, which is not valid.